nixos/vmware-host: Compiler error when compiling kernel module from vmware-modules-workstation

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

An error shows up when compiling a Linux kernel module from the `vmware-modules-workstation` derivation.

### Steps to reproduce

1. Setup a NixOS machine.
2. Add `virtualisation.vmware.host.enable = true;` to your configuration
3. Rebuild
4. Follow the instructions to download the VMware bundle : 
    a. Connect to your Broadcom account and download the latest Linux x64 bundle for VMware Workstation.
    b. Add it to the Nix store using the commands provided
5. Rebuild the configuration, and read the error.

### Expected behaviour

The module builds correctly, without errors.

### Screenshots

_No response_

### Relevant log output

```console
[leo@nixos:~]$ nix log /nix/store/ds9x6i5mp7q1h9fldvfyy48hyvkg7xrr-vmware-modules-workstation-17.6.3-20250608-6.16.drv --log-format raw | tee
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/mmvjc142cnp44cr173g58lb8dfkii5nk-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation vmware-modules-workstation-17.6.3-20250608-6.16: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'Makefile')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j24 SHELL=/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash
make -C vmmon-only 
make -C vmnet-only 
make[1]: Entering directory '/build/source/vmnet-only'
make[1]: Entering directory '/build/source/vmmon-only'
Using kernel build system.
Using kernel build system.
make -C /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build/include/.. M=$PWD SRCROOT=$PWD/. \
  MODULEBUILDDIR= modules
make[2]: Entering directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make[3]: Entering directory '/build/source/vmmon-only'
make -C /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build/include/.. M=$PWD SRCROOT=$PWD/. \
  MODULEBUILDDIR= modules
make[2]: Entering directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make[3]: Entering directory '/build/source/vmnet-only'
  CC [M]  linux/driver.o
  CC [M]  linux/driverLog.o
  CC [M]  linux/hostif.o
  CC [M]  common/apic.o
  CC [M]  common/comport.o
  CC [M]  common/cpuid.o
  CC [M]  common/crosspage.o
  CC [M]  common/memtrack.o
  CC [M]  common/moduleloop.o
  CC [M]  common/phystrack.o
  CC [M]  common/sharedAreaVmmon.o
  CC [M]  common/statVarsVmmon.o
  CC [M]  common/task.o
  CC [M]  common/vmx86.o
  CC [M]  bootstrap/bootstrap.o
  CC [M]  bootstrap/monLoader.o
  CC [M]  bootstrap/monLoaderVmmon.o
  CC [M]  bootstrap/vmmblob.o
In file included from /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/kernel.h:28,
                 from /build/source/vmmon-only/./include/vm_assert.h:77,
                 from /build/source/vmmon-only/./include/x86cpuid.h:39,
                 from /build/source/vmmon-only/./include/x86cpuid_asm.h:39,
                 from common/apic.c:23:
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:314:9: warning: "MIN" redefined
  314 | #define MIN(a, b) __cmp(min, a, b)
      |         ^~~
In file included from common/apic.c:20:
/build/source/vmmon-only/./include/vm_basic_defs.h:97:9: note: this is the location of the previous definition
   97 | #define MIN(_a, _b)   (((_a) < (_b)) ? (_a) : (_b))
      |         ^~~
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:315:9: warning: "MAX" redefined
  315 | #define MAX(a, b) __cmp(max, a, b)
      |         ^~~
/build/source/vmmon-only/./include/vm_basic_defs.h:110:9: note: this is the location of the previous definition
  110 | #define MAX(_a, _b)   (((_a) > (_b)) ? (_a) : (_b))
      |         ^~~
  CC [M]  driver.o
  CC [M]  hub.o
  CC [M]  userif.o
  CC [M]  netif.o
  CC [M]  bridge.o
In file included from /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/kernel.h:28,
                 from /build/source/vmmon-only/./include/vm_assert.h:77,
                 from /build/source/vmmon-only/./include/rateconv.h:44,
                 from /build/source/vmmon-only/./include/ptsc.h:36,
                 from /build/source/vmmon-only/./include/modulecall.h:34,
                 from common/crosspage.c:58:
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:314:9: warning: "MIN" redefined
  314 | #define MIN(a, b) __cmp(min, a, b)
      |         ^~~
In file included from /build/source/vmmon-only/./include/cpu_types.h:29,
                 from /build/source/vmmon-only/./include/modulecall.h:33:
/build/source/vmmon-only/./include/vm_basic_defs.h:97:9: note: this is the location of the previous definition
   97 | #define MIN(_a, _b)   (((_a) < (_b)) ? (_a) : (_b))
      |         ^~~
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:315:9: warning: "MAX" redefined
  315 | #define MAX(a, b) __cmp(max, a, b)
      |         ^~~
/build/source/vmmon-only/./include/vm_basic_defs.h:110:9: note: this is the location of the previous definition
  110 | #define MAX(_a, _b)   (((_a) > (_b)) ? (_a) : (_b))
      |         ^~~
  CC [M]  procfs.o
  CC [M]  smac_compat.o
In file included from /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/kernel.h:28,
                 from /build/source/vmmon-only/./include/vm_assert.h:77,
                 from /build/source/vmmon-only/./include/rateconv.h:44,
                 from /build/source/vmmon-only/./include/ptsc.h:36,
                 from /build/source/vmmon-only/./include/modulecall.h:34,
                 from common/vmx86.h:34,
                 from common/hostif.h:33,
                 from common/memtrack.c:90:
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:314:9: warning: "MIN" redefined
  314 | #define MIN(a, b) __cmp(min, a, b)
      |         ^~~
In file included from /build/source/vmmon-only/./include/cpu_types.h:29,
                 from /build/source/vmmon-only/./include/modulecall.h:33:
/build/source/vmmon-only/./include/vm_basic_defs.h:97:9: note: this is the location of the previous definition
   97 | #define MIN(_a, _b)   (((_a) < (_b)) ? (_a) : (_b))
      |         ^~~
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:315:9: warning: "MAX" redefined
  315 | #define MAX(a, b) __cmp(max, a, b)
      |         ^~~
/build/source/vmmon-only/./include/vm_basic_defs.h:110:9: note: this is the location of the previous definition
  110 | #define MAX(_a, _b)   (((_a) > (_b)) ? (_a) : (_b))
      |         ^~~
In file included from /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/kernel.h:28,
                 from /build/source/vmmon-only/./include/vm_assert.h:77,
                 from /build/source/vmmon-only/./include/rateconv.h:44,
                 from /build/source/vmmon-only/./include/ptsc.h:36,
                 from /build/source/vmmon-only/./include/modulecall.h:34,
                 from common/vmx86.h:34,
                 from common/phystrack.c:45:
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:314:9: warning: "MIN" redefined
  314 | #define MIN(a, b) __cmp(min, a, b)
      |         ^~~
In file included from /build/source/vmmon-only/./include/cpu_types.h:29,
                 from /build/source/vmmon-only/./include/modulecall.h:33:
/build/source/vmmon-only/./include/vm_basic_defs.h:97:9: note: this is the location of the previous definition
   97 | #define MIN(_a, _b)   (((_a) < (_b)) ? (_a) : (_b))
      |         ^~~
/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/include/linux/minmax.h:315:9: warning: "MAX" redefined
  315 | #define MAX(a, b) __cmp(max, a, b)
      |         ^~~
/build/source/vmmon-only/./include/vm_basic_defs.h:110:9: note: this is the location of the previous definition
  110 | #define MAX(_a, _b)   (((_a) > (_b)) ? (_a) : (_b))
      |         ^~~
  CC [M]  smac.o
  CC [M]  vnetEvent.o
  CC [M]  vnetUserListener.o
driver.c:288:1: warning: no previous prototype for 'vmnet_init_module' [-Wmissing-prototypes]
  288 | vmnet_init_module(void)
      | ^~~~~~~~~~~~~~~~~
driver.c:367:1: warning: no previous prototype for 'vmnet_cleanup_module' [-Wmissing-prototypes]
  367 | vmnet_cleanup_module(void)
      | ^~~~~~~~~~~~~~~~~~~~
linux/hostif.c: In function 'HostIF_SafeRDMSR':
linux/hostif.c:3418:10: error: implicit declaration of function 'rdmsrl_safe'; did you mean 'rdmsrq_safe'? [-Wimplicit-function-declaration]
 3418 |    err = rdmsrl_safe(msr, &v);
      |          ^~~~~~~~~~~
      |          rdmsrq_safe
make[5]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/scripts/Makefile.build:287: linux/hostif.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[4]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:2003: .] Error 2
make[3]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
make[3]: Leaving directory '/build/source/vmmon-only'
make[2]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make[1]: *** [Makefile:117: vmmon.ko] Error 2
make[1]: Leaving directory '/build/source/vmmon-only'
make: *** [Makefile:21: vmmon-only] Error 2
make: *** Waiting for unfinished jobs....
  LD [M]  vmnet.o
  MODPOST Module.symvers
  CC [M]  vmnet.mod.o
  CC [M]  .module-common.o
  LD [M]  vmnet.ko
  BTF [M] vmnet.ko
Skipping BTF generation for vmnet.ko due to unavailability of vmlinux
make[3]: Leaving directory '/build/source/vmnet-only'
make[2]: Leaving directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make -C $PWD SRCROOT=$PWD/. \
  MODULEBUILDDIR= postbuild
make[2]: Entering directory '/build/source/vmnet-only'
make[2]: 'postbuild' is up to date.
make[2]: Leaving directory '/build/source/vmnet-only'
cp -f vmnet.ko ./../vmnet.o
make[1]: Leaving directory '/build/source/vmnet-only'

[leo@nixos:~]$ sudo nixos-rebuild switch 
building the system configuration...
error: builder for '/nix/store/ds9x6i5mp7q1h9fldvfyy48hyvkg7xrr-vmware-modules-workstation-17.6.3-20250608-6.16.drv' failed with exit code 2;
       last 25 log lines:
       > make[4]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:2003: .] Error 2
       > make[3]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
       > make[3]: Leaving directory '/build/source/vmmon-only'
       > make[2]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
       > make[2]: Leaving directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
       > make[1]: *** [Makefile:117: vmmon.ko] Error 2
       > make[1]: Leaving directory '/build/source/vmmon-only'
       > make: *** [Makefile:21: vmmon-only] Error 2
       > make: *** Waiting for unfinished jobs....
       >   LD [M]  vmnet.o
       >   MODPOST Module.symvers
       >   CC [M]  vmnet.mod.o
       >   CC [M]  .module-common.o
       >   LD [M]  vmnet.ko
       >   BTF [M] vmnet.ko
       > Skipping BTF generation for vmnet.ko due to unavailability of vmlinux
       > make[3]: Leaving directory '/build/source/vmnet-only'
       > make[2]: Leaving directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
       > make -C $PWD SRCROOT=$PWD/. \
       >   MODULEBUILDDIR= postbuild
       > make[2]: Entering directory '/build/source/vmnet-only'
       > make[2]: 'postbuild' is up to date.
       > make[2]: Leaving directory '/build/source/vmnet-only'
       > cp -f vmnet.ko ./../vmnet.o
       > make[1]: Leaving directory '/build/source/vmnet-only'
       For full logs, run:
         nix log /nix/store/ds9x6i5mp7q1h9fldvfyy48hyvkg7xrr-vmware-modules-workstation-17.6.3-20250608-6.16.drv
error: 1 dependencies of derivation '/nix/store/gi2kpx351igcms0b5jxkngipvfy0brwb-linux-6.16-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w5ah4cmxgg7g131gk31gxwbz7vyxk3fr-linux-6.16-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4kcp346h825b2h4frjlzbp2hyg8w28b7-nixos-system-nixos-25.11.20250731.94def63.drv' failed to build
Command 'nix --extra-experimental-features 'nix-command flakes' build --print-out-paths '/etc/nixos#nixosConfigurations."nixos".config.system.build.toplevel' --no-link' returned non-zero exit status 1.
```

### Additional context

_No response_

### System metadata

[leo@nixos:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.0, NixOS, 25.11 (Xantusia), 25.11.20250731.94def63`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/9x1a429wj6sy54bfr028820n5mgyvdri-source`

### Notify maintainers

@cawilliamson  @deinferno @vifino

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/vmware/default.nix
+++ b/pkgs/os-specific/linux/vmware/default.nix
@@ -9,13 +9,13 @@
 
 stdenv.mkDerivation {
   pname = "vmware-modules";
-  version = "workstation-17.6.3-20250608-${kernel.version}";
+  version = "workstation-17.6.3-20250728-${kernel.version}";
 
   src = fetchFromGitHub {
     owner = "philipl";
     repo = "vmware-host-modules";
-    rev = "e08764f84df3bd14407291e02346b7c5c54d0bf6";
-    hash = "sha256-RDVKropC+VOv1uJijOGlLt6iqZI8Cm9aWlbmvZj2CSk=";
+    rev = "6797e552638a28d1fa1e9ebd7ab5d3c628671ba0";
+    hash = "sha256-KCLxAF6UtNIdKcDoANviln2RJuz1Ld8jq5QFW9ONghs=";
   };
 
   hardeningDisable = [ "pic" ];

